app/controllers/concerns/client_sortable.rb

```diff
@@ -24,38 +24,52 @@ def filtered_and_sorted_clients(default_order: nil)
 
   # see if there are any overlapping keys in the provided params and search/sort set
   def has_search_and_sort_params?
-    (params.keys.map(&:to_sym) & search_and_sort_params).any?
+    overlapping_keys = (params.keys.map(&:to_sym) & search_and_sort_params)
+    hash_params = params.try(:to_unsafe_h) || params
```

Comment:
I'd like to see a test covering these changes. From looking at it, I don't immediately understand the purpose of `params.try(:to_unsafe_h)`. Also, that sounds scary!

Reply:
I think this would be easier to read with a explicit if.

```
if overlapping_keys.any?
  params.require(*overlapping_keys)
end
```

Which I guess is the same as

`params.require(*overlapping_keys) if overlapping_keys.any?`

I'm not 100% sure that my proposed use of `require(*...)` is the same as your use of `.slice`. I am fearful when I see `:to_unsafe_h` so I'm hopeful there's a way to avoid it.